[CloudDirectory] New Resource support

zkarpinski opened the pull request:
Resolves #8703 

- CreateDirectory
- GetDirectory
- DeleteDirectory
- ListDirectories
- TagResource
- UnTagResource
- Added to ResourceGroupTaggingAPI

bblommers commented on moto/clouddirectory/responses.py:
```diff
@@ -0,0 +1,93 @@
+            state=state,
+        )
+        directory_list = [directory.to_dict() for directory in directories]
+        return json.dumps(dict(Directories=directory_list, NextToken=next_token))
```
This should only return a `NextToken` if pagination is actually implemented - otherwise boto3 will end up in an endless loop when using `boto3_client.get_paginator('list_directories')`

bblommers commented on moto/resourcegroupstaggingapi/models.py:
```diff
@@ -265,6 +266,22 @@ def format_tag_keys(
+                        continue
+                    yield {"ResourceARN": f"{directory.directory_arn}", "Tags": tags}
+            except Exception:
+                pass
```
What kind of exception can happen here, and why do we ignore it? Note that I don't mind - but it would be nice to have a comment explaining what's happening for future reference :slightly_smiling_face:

bblommers replied:
Ah, that makes sense, if the `CloudDirectory` service is not available in that region.

It's probably cleaner to use the example from `LexV2` then, with a custom method that returns  `None` if no region exists. Then the whole section can be if-guarded instead, to make sure it only runs when `CloudDirectory` is available.

zkarpinski commented:
Corrections made: 
Updated resourcegroupstaggingapi logic to handle the regions for cloud-directory
Updated pagination logic & states filter for list_

bblommers reviewed (approved):
Great - thanks for the fixes @zkarpinski!
